operator/webhooks/pod_mutating_webhook.go

```diff
@@ -132,21 +131,17 @@ func (a *PodMutatingWebhook) handleWorkload(ctx context.Context, logger logr.Log
 			logger.Error(err, "Could not send workload created K8s event")
 			return err
 		}
-
 		return nil
 	}
 
 	if err != nil {
 		return fmt.Errorf("could not fetch WorkloadInstance"+": %+v", err)
 	}
 
-	if workload.Spec.Version == pod.Annotations[common.VersionAnnotation] {
-		return nil
+	if !reflect.DeepEqual(workload.Spec, newWorkload.Spec) {
+		workload.Spec = newWorkload.Spec
 	}
 
-	workload.Spec.Version = pod.Annotations[common.VersionAnnotation]
-	workload.Spec.ResourceReference = a.getResourceReference(pod)
-	workload.Annotations[common.VersionAnnotation] = pod.Annotations[common.VersionAnnotation]
 	err = a.Client.Update(ctx, workload)
```

Comment:
in this case we are updating the Workload even if nothing changed, we should avoid that